### PR TITLE
Some `async` pipe refactoring

### PR DESCRIPTION
**Pull request by JoostK**
See individual commits

### CODE REVIEW

File: packages/common/src/pipes/async_pipe.ts [hunk 1]
```diff
@@ -82,18 +75,26 @@ const _subscribableStrategy = new SubscribableStrategy();
...
 
   private _subscription: Unsubscribable|Promise<any>|null = null;
   private _obj: Subscribable<any>|Promise<any>|EventEmitter<any>|null = null;
-  private _strategy: SubscriptionStrategy = null!;
+  private _strategy: SubscriptionStrategy|null = null;
 
-  constructor(private _ref: ChangeDetectorRef) {}
+  constructor(ref: ChangeDetectorRef) {
+    this._ref = ref;
+  }
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/async_pipe.ts:87]: nit: I'd propose adding a quick comment to explain why we can't use `private _ref: ChangeDetectorRef` as we had before (due to the typing that should also allow `null`?).

**Review by AndrewKushnir — APPROVED** [review]
LGTM, thanks for the improvement 👍

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api

**Comment by atscott** [comment]
note: blocked by the code freeze for v14

File: packages/common/src/pipes/async_pipe.ts [hunk 2]
```diff
@@ -82,18 +75,28 @@ const _subscribableStrategy = new SubscribableStrategy();
...
+  private _ref: ChangeDetectorRef|null;
   private _latestValue: any = null;
 
   private _subscription: Unsubscribable|Promise<any>|null = null;
   private _obj: Subscribable<any>|Promise<any>|EventEmitter<any>|null = null;
-  private _strategy: SubscriptionStrategy = null!;
+  private _strategy: SubscriptionStrategy|null = null;
 
-  constructor(private _ref: ChangeDetectorRef) {}
+  constructor(ref: ChangeDetectorRef) {
```
Comment (alxhub) [inline packages/common/src/pipes/async_pipe.ts:85]: Alternatively:
```suggestion
  constructor(@Inject(ChangeDetectorRef) private _ref: ChangeDetectorRef|null) {
```

Comment (JoostK) [reply packages/common/src/pipes/async_pipe.ts:85]: Actually I want to avoid this, as it allows instantiating `AsyncPipe` without a CDR

**Comment by alxhub** [comment]
This PR was merged into the repository by commit c26b6d9a426a2e2528c36130376ef854d7f39a18.

**Pull request closed without merging**